handlers/zenvia/zenvia.go

```diff
@@ -1,3 +1,232 @@
 package zenvia
 
-/* no logs */
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/buger/jsonparser"
+	"github.com/nyaruka/courier"
+	"github.com/nyaruka/courier/handlers"
+	"github.com/nyaruka/courier/utils"
+	"github.com/pkg/errors"
+)
+
+var sendURL = "https://api-rest.zenvia360.com.br/services"
+
+func init() {
+	courier.RegisterHandler(NewHandler())
+}
+
+type handler struct {
+	handlers.BaseHandler
+}
+
+// NewHandler returns a new Zenvia handler
+func NewHandler() courier.ChannelHandler {
+	return &handler{handlers.NewBaseHandler(courier.ChannelType("ZV"), "Zenvia")}
+}
+
+// {
+//     "callbackMoRequest": {
+// 	    	"id": "20690090",
+//         	"mobile": "555191951711",
+//         	"shortCode": "40001",
+//         	"account": "zenvia.envio",
+//         	"body": "Content of reply SMS",
+//         	"received": "2014-08-26T12:27:08.488-03:00",
+//         	"correlatedMessageSmsId": "hs765939061"
+//  	}
+// }
+type messageRequest struct {
+	CallbackMoRequest struct {
+		ID         string `validate:"required" json:"id"`
+		From       string `validate:"required" json:"mobile"`
+		Text       string `validate:"required" json:"body"`
+		Date       string `validate:"required" json:"received"`
+		ExternalID string `validate:"required" json:"correlatedMessageSmsId"`
+	}
+}
+
+// {
+// 		"callbackMtRequest": {
+//      	"status": "03",
+//         	"statusMessage": "Delivered",
+//         	"statusDetail": "120",
+//         	"statusDetailMessage": "Message received by mobile",
+//         	"id": "hs765939216",
+//         	"received": "2014-08-26T12:55:48.593-03:00",
+//         	"mobileOperatorName": "Claro"
+// 		}
+// }
+type statusRequest struct {
+	CallbackMtRequest struct {
```

Comment:
Which part? Just saying class names shouldn't have mixed acronym case in Go. So instead of `Mt` you should always use `MT`, just like for `http` should always be `HTTP`